new-lamassu-admin/src/pages/Customers/components/CustomInfoRequestsData.js

```diff
@@ -0,0 +1,198 @@
+    display: 'flex'
+  },
+  disabledBtn: {
+    backgroundColor: disabledColor,
+    '&:hover': {
+      backgroundColor: disabledColor
+    }
+  },
+  linkIcon: {
+    marginTop: 12,
+    marginLeft: 4,
+    cursor: 'pointer'
+  }
+}
+
+const SET_AUTHORIZED_REQUEST = gql`
+  mutation setAuthorizedCustomRequest(
+    $customerId: ID!
+    $infoRequestId: ID!
+    $isAuthorized: Boolean!
+  ) {
+    setAuthorizedCustomRequest(
+      customerId: $customerId
+      infoRequestId: $infoRequestId
+      isAuthorized: $isAuthorized
+    )
+  }
+`
+
+const useStyles = makeStyles(styles)
+const CustomInfoRequestsData = ({ data }) => {
+  const classes = useStyles()
+  const [toView, setToView] = useState(null)
+  const [setAuthorized] = useMutation(SET_AUTHORIZED_REQUEST, {
+    onError: () => console.error('Error while clearing notification'),
+    refetchQueries: () => ['customer']
+  })
+
+  const authorize = it =>
+    setAuthorized({
+      variables: {
+        customerId: it.customerId,
+        infoRequestId: it.customInfoRequest.id,
+        isAuthorized: true
+      }
+    })
+  const reject = it =>
+    setAuthorized({
+      variables: {
+        customerId: it.customerId,
+        infoRequestId: it.customInfoRequest.id,
+        isAuthorized: false
+      }
+    })
+
+  const getBtnClasses = (it, isAuthorize) => {
+    return {
+      [classes.actionButton]: true,
+      [classes.disabledBtn]:
+        (isAuthorize && it.approved === true) ||
+        (!isAuthorize && it.approved === false)
+    }
+  }
+
+  const AuthorizeButton = it => (
+    <ActionButton
+      className={classnames(getBtnClasses(it, true))}
+      color="secondary"
+      Icon={AuthorizeIcon}
+      InverseIcon={AuthorizeReversedIcon}
+      onClick={() => authorize(it)}>
+      Authorize
+    </ActionButton>
+  )
+
+  const RejectButton = it => (
+    <ActionButton
+      className={classnames(getBtnClasses(it, false))}
+      color="secondary"
+      Icon={RejectIcon}
+      InverseIcon={RejectReversedIcon}
+      onClick={() => reject(it)}>
+      Reject
+    </ActionButton>
+  )
+
+  const getActionButtons = it => {
+    return (
+      <>
+        {AuthorizeButton(it)}
+        {RejectButton(it)}
+      </>
+    )
+  }
+
+  const getAuthorizedStatus = it => {
+    return it.approved === null
+      ? { label: 'Pending', type: 'neutral' }
+      : it.approved === false
+      ? { label: 'Rejected', type: 'error' }
+      : { label: 'Accepted', type: 'success' }
+  }
+
+  return (
+    <>
+      <H3>Custom Info Requests Data</H3>
+      <div>
+        <Table>
+          <THead>
+            <Th width={250}>Custom Request Name</Th>
+            <Th width={500}>Custom Request Data</Th>
+            <Th width={200}>Status</Th>
+            <Th width={250} textAlign="center">
+              Actions
+            </Th>
+          </THead>
+          <TBody>
+            {data.map((it, idx) => {
+              return (
+                <React.Fragment hey={idx}>
```

Comment:
Is this `hey` used anywhere? Feels like a development leftover or a misspelled `key` prop

Reply:
haha yep